### PR TITLE
feat: get events endpoint

### PR DESCRIPTION
**Pull request by Iloniavo**

### CODE REVIEW

File: doc/api.yml [hunk 1]
```diff
@@ -2477,6 +2471,23 @@ paths:
...
+          schema:
+            $ref: '#/components/schemas/Page'
+          description: Set value to 1 by default if null is provided
+          required: false
+        - name: page_size
+          in: query
+          description: Set value to 15 by default if null is provided
+          schema:
+            $ref: '#/components/schemas/PageSize'
+        - name: group_id
```
Comment (Mahefaa) [inline doc/api.yml]: group_ref, filters are made for humans, so we put easy values

File: doc/api.yml [hunk 2]
```diff
@@ -2506,7 +2517,7 @@ paths:
           schema:
             type: string
           required: true
-      operationId: updateEventParticipantsStatus
+      operationId: crupdateEventParticipants
```
Comment (Mahefaa) [inline doc/api.yml]: does it CREATE or UPDATE ?

File: doc/api.yml [hunk 3]
```diff
@@ -3161,13 +3172,24 @@ components:
...
         event_status:
           $ref: '#/components/schemas/AttendanceStatus'
+        group_id:
+          type: string
+    EventParticipant:
+      allOf:
+        - $ref: '#/components/schemas/UserIdentifier'
+      type: object
+      properties:
+        group:
```
Comment (Mahefaa) [inline doc/api.yml]: group_name ? or ref ?

File: src/main/java/school/hei/haapi/endpoint/rest/controller/EventController.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  @GetMapping("/events")
+  public List<Event> getEvents(
+      @RequestParam(name = "page") PageFromOne page,
+      @RequestParam(name = "page_size") BoundedPageSize pageSize,
+      @RequestParam(name = "planner_name", required = false) String plannerName,
+      @RequestParam(required = false) Instant from,
+      @RequestParam(required = false) Instant to) {
+    return eventService.getEvents(plannerName, from, to, page, pageSize).stream()
+        .map(mapper::toRest)
+        .collect(Collectors.toUnmodifiableList());
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/endpoint/rest/controller/EventController.java]: static import

File: src/main/java/school/hei/haapi/model/EventParticipant.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  @ManyToOne
+  @JoinColumn(name = "participant_id", referencedColumnName = "id")
+  private User participant;
+
+  @JdbcTypeCode(NAMED_ENUM)
+  @Enumerated(STRING)
+  private AttendanceStatus status;
+
+  @ManyToOne
+  @JoinColumn(name = "actual_group_id", referencedColumnName = "id")
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/model/EventParticipant.java]: why the need for "actual"

File: src/main/java/school/hei/haapi/repository/GroupRepository.java [hunk 6]
```diff
@@ -11,4 +12,6 @@
 public interface GroupRepository extends JpaRepository<Group, String> {
   @Query("select g from Group g")
   Page<Group> getGroups(Pageable pageable);
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/repository/GroupRepository.java]: change this. you do not need to such easy Query as @Query

File: src/main/java/school/hei/haapi/service/EventService.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                    user -> {
+                      eventParticipants.add(
+                          EventParticipant.builder()
+                              .event(event)
+                              .participant(user)
+                              .status(MISSING)
+                              .group(actualGroup)
+                              .build());
+                    });
+                eventParticipantService.crupdateEventParticipants(eventParticipants);
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/service/EventService.java]: put in a public method

File: src/main/resources/db/migration/V0_2__Create_user_table.sql [hunk 8]
```diff
@@ -1,4 +1,4 @@
-do
+    do
```
Comment (Mahefaa) [inline src/main/resources/db/migration/V0_2__Create_user_table.sql]: why indent AND MOSTLY why indent V0_2 ?

File: src/main/resources/db/migration/V42_19__Create_event_participant_table.sql [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+end if;
+end
+$$;
+
+create table if not exists "event_participant"(
+    id varchar constraint event_participant_pk primary key default uuid_generate_v4(),
+    event_id varchar constraint event_id_fk references "event"(id),
+    participant_id varchar constraint participant_id_fk references "user"(id),
+    status attendance_status default 'MISSING',
+    actual_group_id varchar constraint event_participant_group_id_fk references "group"(id)
```
Comment (Mahefaa) [inline src/main/resources/db/migration/V42_19__Create_event_participant_table.sql]: no need for 'actual' suffix

**Review by Mahefaa — CHANGES_REQUESTED** [review]
add event filter by type also

File: doc/api.yml (hunk 3 above)
Comment (Iloniavo) [reply doc/api.yml]: Group name

File: doc/api.yml [hunk 10]
```diff
@@ -2499,14 +2510,14 @@ paths:
     put:
       tags:
         - Events
-      summary: Update event participants status
+      summary: Create or pdate event participants
```
Comment (Mahefaa) [inline doc/api.yml]: typo: pdate -> update

File: doc/api.yml [hunk 11]
```diff
@@ -2477,6 +2471,23 @@ paths:
...
+          description: Set value to 1 by default if null is provided
+          required: false
+        - name: page_size
+          in: query
+          description: Set value to 15 by default if null is provided
+          schema:
+            $ref: '#/components/schemas/PageSize'
+        - name: group_ref
+          in: query
+          description: Group ref of the participant
```
Comment (Mahefaa) [inline doc/api.yml]: filter by group_ref

Comment (Mahefaa) [reply doc/api.yml]: it is a better description

File: src/main/java/school/hei/haapi/endpoint/rest/controller/EventController.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+  private final EventParticipantService eventParticipantService;
+
+  @PutMapping("/events")
+  public List<Event> crupdateEvents(@RequestBody List<CreateEvent> eventsToSave) {
+    return eventService
+        .createOrUpdateEvent(eventsToSave.stream().map(mapper::toDomain).toList())
+        .stream()
+        .map(mapper::toRest)
+        .collect(toUnmodifiableList());
+  }
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/endpoint/rest/controller/EventController.java:41]: group put /events and group /events

**Review by Mahefaa — CHANGES_REQUESTED** [review]

File: src/main/java/school/hei/haapi/model/Event.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+  @Enumerated(STRING)
+  private EventType type;
+
+  private String description;
+
+  @Column(name = "begin_datetime")
+  private Instant begin;
+
+  @Column(name = "end_datetime")
+  private Instant end;
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/model/Event.java]: use endDatetime as variable name, same for begin -> beginDatetime etc ( for api also)

**Review by Mahefaa — CHANGES_REQUESTED** [review]

File: doc/api.yml [hunk 14]
```diff
@@ -2506,7 +2517,7 @@ paths:
           schema:
             type: string
           required: true
-      operationId: updateEventParticipantsStatus
+      operationId: updateEventParticipants
```
Comment (Mahefaa) [inline doc/api.yml]: you only update statuses so keep updateEventParticipantsStatus

File: doc/api.yml [hunk 15]
```diff
@@ -2515,7 +2526,7 @@ paths:
             schema:
               type: array
               items:
-                $ref: '#/components/schemas/GroupedEventParticipant'
+                $ref: '#/components/schemas/CrupdateEventParticipant'
```
Comment (Mahefaa) [inline doc/api.yml]: the component doesn't Crupdate

**Review by Mahefaa — CHANGES_REQUESTED** [review]

**Pull request merged**